packages/muze-legend/src/legend/discrete-legend.js

```diff
@@ -45,18 +50,23 @@ export default class DiscreteLegend extends SimpleLegend {
     dataFromScale (scale) {
         const { scaleType, domain, scaleFn } = getScaleInfo(scale);
         let domainForLegend = [...new Set(domain)];
-        const type = this.metaData().getData().schema[0].type;
+        const field = this.metaData().getFieldspace().fields[0];
+        const { type, subtype } = field.schema();
 
         domainForLegend = domainForLegend.map((ele, i) => {
             let value = 0;
             let range = 0;
-            if (type === 'measure') {
+            if (type === FieldType.MEASURE) {
```

Comment:
If this is a discrete legend then it will never get created if a measure is assigned to color